src/tableView.js

```diff
@@ -364,6 +383,49 @@ class TableView {
     return Math.min(this.instance.columnIndexMapper.getRenderableIndexesLength(), this.settings.maxCols);
   }
 
+  /**
+   * Returns the number of renderable rows.
+   *
+   * @returns {number}
+   */
+  countRenderableRows() {
+    return Math.min(this.instance.rowIndexMapper.getRenderableIndexesLength(), this.settings.maxRows);
+  }
+
+  /**
+   * Returns number of not hidden records counting from the passed starting index.
+   * The counting direction can be controlled by `incrementBy` argument.
+   *
+   * @param {number} visualIndex The visual index from which the counting begins.
+   * @param {string} axix The axis as 'row' or 'column'.
+   * @param {number} incrementBy If `-1` then counting is backwards or forward when `1`.
+   * @returns {number}
+   */
+  countNotHiddenRecords(visualIndex, axix, incrementBy) {
+    if (isNaN(visualIndex)) {
+      return 0;
+    }
+
+    const indexMapper = this.instance[`${axix}IndexMapper`];
```

Comment:
Fixed within https://github.com/handsontable/handsontable/pull/6887/commits/059c37ee5fd864e2458f91d9ca0009f2abb13031.

Reply:
What about using the reference: https://github.com/handsontable/handsontable/pull/6887#discussion_r427271670? It's an easy way and safe.

Reply:
The `countNotHiddenRecords` is a method of the TableView class, not a pure function which acts as utils. I don't see anything wrong that the IndexMapper instance is used within the method body. I'd leave as it is.

Reply:
@wszymanski `axis` is used to create index mapper reference AND method name to get a count. So using reference is not enough to refactor this method if we want to avoid dynamic keys.

Idea 1:
Make the `countNotHiddenRecords` private and introduce two public methods that will use it internally. The pro here is that we reduce the probability of incorrect input and therefore throwing an error
```ts
  countNotHiddenRows(visualIndex, incrementBy) {
    return this.countNotHiddenRecords(visualIndex, 'row', incrementBy);
  }

  countNotHiddenColumns(visualIndex, incrementBy) {
    return this.countNotHiddenRecords(visualIndex, 'column', incrementBy);
  }

  /** @private */
  countNotHiddenRecords(visualIndex, axis, incrementBy) {
    //...
  }
```

Idea 2:
Instead of creating dynamic key we can use ternary:
```diff
- const indexMapper = this.instance[`${axis}IndexMapper`];
+ const indexMapper = axis === 'row' ? this.instance.rowIndexMapper : this.instance.columnIndexMapper;
```
and
```diff
- notHiddenRows = this[`countRenderable${toUpperCaseFirst(axix)}s`]() - renderableIndex;
+ const count = axis === 'row' ? this.countRenderableRows() : this.countRenderableColumns();
+ notHiddenRows = this[`countRenderable${toUpperCaseFirst(axix)}s`]() - renderableIndex;
```
I would use both

Reply:
I always try to avoid that kind of "incomplete" expressions:
```
const indexMapper = axis === 'row' ? this.instance.rowIndexMapper : this.instance.columnIndexMapper;
```
Here I see a potentially silent error where runtime JS can't catch. For instance, when the axis will be passed as `rOw` it will be fallback to column mapper.

In this expression:
```
const indexMapper = this.instance[`${axis}IndexMapper`];
```
Passing the wrong axis throws kind of TypeError as soon as code bellow try to access to the instance.

Reply:
I agree with @wojciechczerniak.

Reply:
Fixed within https://github.com/handsontable/handsontable/pull/6887/commits/a88b879f63d34a710a2f68d9e524b1ddeea6d568.